lib/jekyll_pages_api/generator.rb

```diff
@@ -10,7 +12,12 @@ def initialize(site)
     end
 
     def pages
-      self.site.pages.map{|page| Page.new(page) }.select(&:html?)
+      result = Array.new
+      self.site.each_site_file do |p|
```

Comment:
Yeah, I suppose it is. :-) This is the current implementation:

```ruby
    def each_site_file
      %w(posts pages static_files docs_to_write).each do |type|
        send(type).each do |item|
          yield item
        end
      end
    end
```
`posts`, `pages`, and `static_files` are listed with `attr_accessor`. `docs_to_write` is a method that runs through collection documents. We can define our own method instead of using `each_site_file`, or keep it for now and fix it if it ever changes, which shouldn't require major surgery.